storage/innobase/include/row0upd.h

```diff
@@ -573,6 +573,10 @@ struct upd_node_t{
 				compilation; speeds up execution:
 				UPD_NODE_NO_ORD_CHANGE and
 				UPD_NODE_NO_SIZE_CHANGE, ORed */
+	/** working with system versioned table */
+	bool		versioned;
+	/** set sys_trx_end = CUR_TRX_ID */
+	bool		vers_delete;
```

Comment:
> We do not want to test the vers_set_fields condition for every inserted row. Instead, we need it once per statement. It can be computed in build_template() or some related function.

Answered [here](https://github.com/MariaDB/server/pull/478#discussion_r152949147)